sdk/dotnet/Pulumi/Deployment/Deployment_RootResource.cs

```diff
@@ -23,21 +24,18 @@ public partial class Deployment
             if (type == Stack._rootPulumiStackTypeName)
                 return null;
 
-            if (_rootResource == null)
-                throw new InvalidOperationException($"Calling {nameof(GetRootResourceAsync)} before the root resource was registered!");
+            lock (_rootResourceLock)
```

Comment:
i don't mind this.  but was this actually necessary?

Reply:
I guess it's not required now, but to tell this I need to trace all the call paths... So I thought a lock would be a sure-fire way to stop bothering.